src/interpreter/ArithmeticHelper.ts

```diff
@@ -695,38 +720,40 @@ function escapeNoCharacters(pattern: string, caseSensitive: boolean): string {
   return str
 }
 
-function inferExtendedNumberTypeAdditive(left: NumberType, right: NumberType): NumberType {
+function inferExtendedNumberTypeAdditive(left: NumberType, right: NumberType, leftFormat: FormatInfo, rightFormat: FormatInfo): [NumberType, FormatInfo] {
```

Comment:
I'm just wondering if it will not be better to pass the whole `rightType` and the `ExtendedNumberFactory` could choose from it what it needs. We would be ready for other properties we chould choose from in the future.

```suggestion
function inferExtendedNumberTypeAdditive(left: RichNumber, right: RichNumber): RichNumber {
  if(left === NumberType.NUMBER_RAW) {
    return right
  }
```

Reply:
Then it's up to the `ExtendedNumberFactory` to decide if there are any additional properties to choose from:
```ts
public ExtendedNumberFactory(value: number, type: RichNumber): ExtendedNumber {
  const { numberType, format, futureProp } = type;
```

Reply:
WDYT?

Reply:
we not always have a RichNumber lying around to pass it as an example, and creating one artificially just to pass the type to the function seems.. artificial. I'll refactor it in some way into that direction.

Reply:
Right! It might be any type that have `format` property or others. But if we have to artificially pass something weird it may not be worth to do it now. We may wait for a real use case.

Reply:
should be better now